12-offer-encoding.md

```diff
@@ -238,296 +233,328 @@ The human-readable prefix for offers is `lno`.
 
 A reader of an offer:
-  - if `features` contains unknown _odd_ bits that are non-zero:
+  - if the offer contains any unknown TLV fields greater or equal to 80:
+    - MUST NOT respond to the offer.
+  - if `offer_features` contains unknown _odd_ bits that are non-zero:
     - MUST ignore the bit.
-  - if `features` contains unknown _even_ bits that are non-zero:
+  - if `offer_features` contains unknown _even_ bits that are non-zero:
     - MUST NOT respond to the offer.
     - SHOULD indicate the unknown bit to the user.
-  - if `description` is not set:
+  - if `offer_description` is not set:
     - MUST NOT respond to the offer.
-  - if `node_id` is not set:
-    - if `paths` is not set or does not contain at least one `blinded_path`:
-      - MUST NOT respond to the offer.
-    - MUST use the final `onionmsg_path` `point` in the first `blinded_path` as the `node_id`.
-  - if `signature` is present, but is not a valid signature using `node_id` as described in [Signature Calculation](#signature-calculation):
+  - if `offer_node_id` is not set:
     - MUST NOT respond to the offer.
-  - if it uses `amount` to provide the user with a cost estimate:
+  - if it uses `offer_amount` to provide the user with a cost estimate:
     - MUST warn user if amount of actual invoice differs significantly
         from that expectation.
   - SHOULD not respond to an offer if the current time is after
-    `absolute_expiry`.
+    `offer_absolute_expiry`.
   - FIXME: more!
 
 ## Rationale
 
-A signature is optional because it makes for a longer string (potentially
+The entire offer is reflected in the invoice_request, both for
+completeness (so all information will be returned in the invoice), and
+so that the offer node can be stateless.  This makes `offer_metadata`
+particularly useful, since it can contain an authentication cookie to
+validate the other fields.
+
+A signature is unnecessary, and makes for a longer string (potentially
 limiting QR code use on low-end cameras); if the offer has an error, no
 invoice will be given (or, for `send_invoice` offers, accepted) since
 the `offer_id` already covers all the non-signature fields.
 
-The `node_id` is redundant if a `blinded_path` is provided: a public
-node can always arrange that the final `node_id` in the path is
-unblinded.
-
 # Invoice Requests
 
 Invoice Requests are a request for an invoice; the human-readable prefix for
-invoices is `lnr`.
+invoices is `lnr`.  It mirrors all the fields from the offer, except
+`offer_send_invoice` which cannot cause invoice_requests.
+
+Note: the `invoice_request_payer_info` is numbered 0 (not in the
+80-159 range for other invoice_request fields) as this is the first
+TLV element, which ensures payer-provided entropy is used in hashing
+for [Signature Calculation](#signature-calculation).
 
 ## TLV Fields for `invoice_request`
 
 1. `tlv_stream`: `invoice_request`
 2. types:
-    1. type: 3 (`chain`)
+    1. type: 0 (`invoice_request_payer_info`)
```

Comment:
IIUC, I think a better name would be `invoice_request_metadata`, for uniformity with `offer_metadata`. They seem to serve a similar purpose?

Reply:
Renamed.